bip39-lib/src/nonJvmMain/kotlin/cash/z/ecc/android/crypto/Pbkdf2Sha512.kt

```diff
@@ -0,0 +1,82 @@
+package cash.z.ecc.android.crypto
+
+import okio.Buffer
+import okio.ByteString.Companion.toByteString
+import okio.use
+import kotlin.experimental.xor
+import kotlin.math.ceil
+
+/**
+ *
+ * This is a clean-room implementation of PBKDF2 using RFC 2898 as a reference.
+ *
+ *
+ * RFC 2898: http://tools.ietf.org/html/rfc2898#section-5.2
+ *
+ *
+ * This code passes all RFC 6070 test vectors: http://tools.ietf.org/html/rfc6070
+ *
+ *
+ * http://cryptofreek.org/2012/11/29/pbkdf2-pure-java-implementation/<br></br>
+ * Modified to use SHA-512 - Ken Sedgwick ken@bonsai.com
+ * Modified to for Kotlin - Kevin Gorham anothergmale@gmail.com
+ * Modified to for Kotlin Multiplatform w/ okio - Luca Spinazzola anothergmale@gmail.com
+ */
+internal actual object Pbkdf2Sha512 {
+
+    /**
+     * Generate a derived key from the given parameters.
+     *
+     * @param p the password
+     * @param s the salt
+     * @param c the iteration count
+     * @param dkLen the key length in bits
+     */
+    actual fun derive(p: CharArray, s: ByteArray, c: Int, dkLen: Int): ByteArray {
+        Buffer().use { baos ->
```

Comment:
Is there a way to reduce using Okio for this?

Reply:
removed its use